lab5/DescendentRecursiveParser.cpp

```diff
@@ -0,0 +1,148 @@
+//
+// Created by radoo on 12.12.2022.
+//
+
+#include "DescendentRecursiveParser.h"
+
+void DescendentRecursiveParser::expand(Configuration &config, Grammar grammar,
+                                       const std::vector<std::string> &input) {
+    if (!grammar.isTerminal(config.inputStack.top())) {
+        std::string A = config.inputStack.top();
```

Comment:
config.inputStack.top() extracted in const var before the if, since you use it twice?